### PR TITLE
Allow ocsci-conf read multiple conf files

### PR DESCRIPTION
**Pull request by petr-balogh**
Initial step for this issue #700.

Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

File: ocs_ci/framework/main.py [hunk 1]
```diff
@@ -39,14 +39,23 @@ def init_ocsci_conf(arguments=None):
     custom_config = get_param('--ocsci-conf', arguments)
     cluster_config = get_param('--cluster-conf', arguments)
     if custom_config:
-        with open(os.path.expanduser(custom_config)) as file_stream:
-            custom_config_data = yaml.safe_load(file_stream)
-            framework.config.update(custom_config_data)
+        config_files = [cc.strip() for cc in custom_config.split(",")]
```
Comment (vasukulkarni) [inline ocs_ci/framework/main.py]: I think we meant repeating --ocsci-conf , the docopt equivalent would be [ocsci-conf=conf]...

Comment (petr-balogh) [reply ocs_ci/framework/main.py]: Please see my comments about it here: https://github.com/red-hat-storage/ocs-ci/issues/700 . I don't see much additional value to have the parameter repeatable like `--ocscu-conf conf/1.yaml --ocsci-con conf/2.yaml`   . The one parameter `--ocscu-conf conf/1.yaml,conf/2.yaml` is also easy to use and does not require write `--ocscu-conf` twice.

Comment (vasukulkarni) [reply ocs_ci/framework/main.py]: I can understand if its hard to fix due to our own param handling, but repeatable args are standard and better than comma separated for file names.  It also helps with auto completion on bash shell.

Comment (petr-balogh) [reply ocs_ci/framework/main.py]: yep for auto completion of files is cool thing. I changed the behavior of function to do it like that. The rest is explained in the issue.

**Comment by vasukulkarni** [comment]
LGTM (would have liked if argparse had worked but I think its kind of chicken/egg who takes over)

File: ocs_ci/framework/main.py [hunk 2]
```diff
@@ -36,17 +48,24 @@ def init_ocsci_conf(arguments=None):
...
+            os.path.abspath(os.path.expanduser(config_file))
+        ) as file_stream:
             custom_config_data = yaml.safe_load(file_stream)
             framework.config.update(custom_config_data)
     if cluster_config:
         with open(os.path.expanduser(cluster_config)) as file_stream:
             cluster_config_data = yaml.safe_load(file_stream)
             framework.config.update(cluster_config_data)
     framework.config.RUN['run_id'] = int(time.time())
+    bin_dir = framework.config.RUN.get('bin_dir')
```
Comment (vavuthu) [inline ocs_ci/framework/main.py]: since we are already loading bin-dir in line 81, can we append bin_dir  in the same place ( after line 82 ).

**Comment by dahorak** [comment]
> LGTM (would have liked if argparse had worked but I think its kind of chicken/egg who takes over)

Wouldn't be possible to use argparse and [disable the help output](https://docs.python.org/3/library/argparse.html#add-help)?
```
parser = argparse.ArgumentParser(add_help=False)
```
(I didn't try it together with the pytest code, so not sure if it will work or not.)

File: ocs_ci/framework/main.py [hunk 3]
```diff
@@ -59,6 +78,7 @@ def main(arguments):
         '-p', 'ocs_ci.framework.pytest_customization.reports',
         '--logger-logsdir', pytest_logs_dir,
     ])
-    utils.add_path_to_env_path(os.path.expanduser(
-        framework.config.RUN['bin_dir']))
+    bin_dir = framework.config.RUN.get('bin_dir')
+    if bin_dir:
+        utils.add_path_to_env_path(bin_dir)
```
Comment (clacroix12) [inline ocs_ci/framework/main.py]: Why do we have to add `bin_dir` to the path here when we are already adding the absolute path in `init_ocsci_conf`?

Comment (petr-balogh) [reply ocs_ci/framework/main.py]: We always added bin dir in the PATH. So didn't changed the logic here. Just moved it all to the function init config for now to have it in one place.

**Comment by petr-balogh** [comment]
@dahorak thanks for sharing this functionality, I didn't know about this prase_known_args and option to not process help, aligned the code

File: ocs_ci/framework/main.py [hunk 4]
```diff
@@ -8,45 +9,36 @@
...
-        return
-    custom_config = get_param('--ocsci-conf', arguments)
-    cluster_config = get_param('--cluster-conf', arguments)
-    if custom_config:
-        with open(os.path.expanduser(custom_config)) as file_stream:
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument('--ocsci-conf', action='append', default=[])
+    # cluster-conf parameter will be deleted once we will update all the jobs
+    parser.add_argument('--cluster-conf')
+    args, unknown = parser.parse_known_args()
```
Comment (dahorak) [inline ocs_ci/framework/main.py]: This should fix the failing unit tests.
```suggestion
    args, unknown = parser.parse_known_args(args=arguments)
```

File: ocs_ci/framework/main.py (hunk 2 above)
Comment (petr-balogh) [reply ocs_ci/framework/main.py]: Moved everything here to be on one place

**Comment by petr-balogh** [comment]
Verifying in my temp job with modification to provide config with --ocsci-conf: 
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster-pbalogh-test/8/console

looks like it's running deployment OK with 3 provided config files as you can see from console output.

File: ocs_ci/framework/main.py (hunk 3 above)
Comment (clacroix12) [reply ocs_ci/framework/main.py]: @petr-balogh right, we were doing it in two places when I had commented this.

**Pull request merged**